File: spring/pcbuilder/src/main/java/ar/edu/um/pcbuilder/dtos/entities/sales/SaleDto.java
package ar.edu.um.pcbuilder.dtos.entities.sales;

import java.sql.Timestamp;
import java.util.List;
import lombok.Data;

@Data
public class SaleDto {

  private Long id;
  private Long userId;
  private Long deviceId;
  private Double finalPrice;
  private Timestamp saleDate;
  private List<SalePersonalizationDto> personalizations;
  private List<SaleAdditionalItemDto> additionalItems;
}

File: kmp/composeApp/src/commonMain/kotlin/ar/edu/um/pcbuilder/dtos/entities/sales/SaleDto.kt
package ar.edu.um.pcbuilder.dtos.entities.sales

import kotlinx.datetime.Instant

data class SaleDto(
    val id: Long?,
    val userId: Long?,
    val deviceId: Long?,
    val finalPrice: Double?,
    val saleDate: Instant?,
    val personalizations: List<SalePersonalizationDto>?,
    val additionalItems: List<SaleAdditionalItemDto>?
)